frontend/src/libs/components/link/link.tsx

```diff
@@ -1,21 +1,36 @@
 import { NavLink } from 'react-router-dom';
 
-import { type AppRoute } from '~/libs/enums/enums.js';
+import { type AppRoute, type ArticleSubRoute } from '~/libs/enums/enums.js';
 import { getValidClassNames } from '~/libs/helpers/helpers.js';
+import { useCallback } from '~/libs/hooks/hooks.js';
 import { type ValueOf } from '~/libs/types/types.js';
 
 import styles from './styles.module.scss';
 
 type Properties = {
-  to: ValueOf<typeof AppRoute>;
+  to: ValueOf<typeof AppRoute> | ValueOf<typeof ArticleSubRoute>;
   className?: string;
+  activeClassName?: string;
   children: React.ReactNode;
 };
 
-const Link: React.FC<Properties> = ({ children, className, to }) => (
-  <NavLink to={to} className={getValidClassNames(styles.link, className)}>
-    {children}
-  </NavLink>
-);
+const Link: React.FC<Properties> = ({
+  children,
+  className,
+  activeClassName,
+  to,
+}) => {
+  const handleGetClassesByLinkState = useCallback(
+    ({ isActive }: { isActive: boolean }): string =>
+      getValidClassNames(styles.link, className, isActive && activeClassName),
```

Comment:
Without brackets prettier may shuffle line as it wants (and create unnecessary diff without any reason)
```suggestion
  const handleGetClassesByLinkState = useCallback(
    ({ isActive }: { isActive: boolean }): string => {
      return getValidClassNames(styles.link, className, isActive && activeClassName);
    },
```